File: modules/tools/permissions.go
package tools

// PermissionDecision represents the outcome of a permission check.
type PermissionDecision int

const (
	DecisionAllow       PermissionDecision = iota // explicitly allowed
	DecisionDeny                                  // explicitly denied
	DecisionAsk                                   // ask the user
	DecisionAutoApprove                           // auto-approved
	DecisionConditional                           // allowed with conditions
)

// PermissionContext provides the information needed to make a permission decision.
type PermissionContext struct {
	WorkingDir string // current working directory
	FilePath   string // file being accessed (if applicable)
	Command    string // command being run (if applicable)
	IsReadOnly bool   // whether the operation is read-only
}

File: packages/tools/src/permissions.ts
/**
 * Permission types for tool access control.
 *
 * Ported from modules/tools/permissions.go.
 */

// ---------------------------------------------------------------------------
// PermissionDecision
// ---------------------------------------------------------------------------

export enum PermissionDecision {
  /** Explicitly allowed. */
  Allow = 0,
  /** Explicitly denied. */
  Deny = 1,
  /** Ask the user. */
  Ask = 2,
  /** Auto-approved. */
  AutoApprove = 3,
  /** Allowed with conditions. */
  Conditional = 4,
}

// ---------------------------------------------------------------------------
// PermissionContext
// ---------------------------------------------------------------------------

/** Provides the information needed to make a permission decision. */
export interface PermissionContext {
  /** Current working directory. */
  workingDir: string;
  /** File being accessed (if applicable). */
  filePath?: string;
  /** Command being run (if applicable). */
  command?: string;
  /** Whether the operation is read-only. */
  isReadOnly: boolean;
}
